scripts/zones/Ceizak_Battlegrounds/mobs/Unfettered_Twitherym.lua

```diff
@@ -0,0 +1,119 @@
+    Thunder = { xi.skillchainType.IMPACTION, xi.skillchainType.FRAGMENTATION, xi.skillchainType.LIGHT, xi.skillchainType.LIGHT_II },
+    Wind = { xi.skillchainType.DETONATION, xi.skillchainType.FRAGMENTATION, xi.skillchainType.LIGHT, xi.skillchainType.LIGHT_II },
+    Ice = { xi.skillchainType.INDURATION, xi.skillchainType.DISTORTION, xi.skillchainType.DARKNESS, xi.skillchainType.DARKNESS_II },
+    Earth = { xi.skillchainType.SCISSION, xi.skillchainType.GRAVITATION, xi.skillchainType.DARKNESS, xi.skillchainType.DARKNESS_II },
+    Dark = { xi.skillchainType.COMPRESSION, xi.skillchainType.DARKNESS, xi.skillchainType.DARKNESS_II },
+    Light = { xi.skillchainType.TRANSFIXION, xi.skillchainType.LIGHT, xi.skillchainType.LIGHT_II }
+}
+
+local chosenElement = nil
+local phaseDTApplied = {
+    ['74_49'] = false,
+    ['49_24'] = false,
+    ['24_0'] = false
+}
+
+local function chooseAffinity()
+    local affinities = { 'Fire', 'Water', 'Thunder', 'Wind', 'Ice', 'Earth', 'Dark', 'Light' }
+    chosenElement = utils.randomEntry(affinities)
+end
+
+local function isSkillchainCorrect(skillchainID)
+    local affinitySkillchains = skillchainAffinities[chosenElement]
+    for _, chain in ipairs(affinitySkillchains) do
+        if skillchainID == chain then
+            return true
+        end
+    end
+
+    return false
+end
+
+local function applyDamageReduction(mob, phase)
+    if not phaseDTApplied[phase] then
+        mob:setMod(xi.mod.UDMGPHYS, damageReductionValue)
+        mob:setMod(xi.mod.UDMGMAGIC, damageReductionValue)
+        mob:setMod(xi.mod.UDMGRANGE, damageReductionValue)
+        mob:setMod(xi.mod.UDMGBREATH, damageReductionValue)
+        phaseDTApplied[phase] = true
+    end
+end
+
+local function resetDamageModifiers(mob)
+    for _, mod in ipairs({ xi.mod.UDMGPHYS, xi.mod.UDMGMAGIC, xi.mod.UDMGRANGE, xi.mod.UDMGBREATH }) do
+        mob:setMod(mod, originalDamage)
+    end
+end
+
+local function getCurrentPhase(mob)
+    local hpp = mob:getHPP()
+    if hpp <= 74 and hpp > 49 then
+        return '74_49'
+    elseif hpp <= 49 and hpp > 24 then
+        return '49_24'
+    elseif hpp <= 24 then
+        return '24_0'
+    end
+
+    return nil
+end
+
+entity.onMobFight = function(mob)
+    local phase = getCurrentPhase(mob)
+    if phase then
+        if mob:getHPP() <= 50 then
+            mob:setMobMod(xi.mobMod.SKILL_LIST, 2018)
+        else
+            mob:setMobMod(xi.mobMod.SKILL_LIST, 2017)
+        end
+    end
+end
+
+entity.onMobSpawn = function(mob)
+    originalDamage = mob:getMod(xi.mod.UDMGPHYS)
+
+    mob:addListener('WEAPONSKILL_USE', 'ALL_MOBSKILL_CHECK', function(mobArg, Target, Skillid, TP, action)
+        local phase = getCurrentPhase(mobArg)
+        if phase then
+            applyDamageReduction(mobArg, phase)  -- Always apply damage reduction based on phase
+            if mobArg:getHP() <= mobArg:getMaxHP() * 0.75 then
+                chooseAffinity()
```

Comment:
This is the only place that function is called.

If you want to store the value of the element ID the mob has chosen, you could set a local var on the mob.